mantidproject/mantid
# Indirect - Bayes - Move Run button for disabling while processing

**Description of work.**
This PR moves the run button above the output options on the Indirect Bayes interface.
This PR ensures the run, plot and save options are disabled while running and plotting is taking place.
This PR also tidies up some of the code previously used to disable buttons in other interfaces.

**To test:**
1. `Interfaces`->`Indirect`->`Bayes`
2. Load the data below
3. Click `Run` and ensure the run, plot and save options are disabled
4. Click `Plot` and ensure the run, plot and save options are disabled

- For the third interface `Stretch`, clicking `Plot` should result in an error message. This is a good way of checking that all the buttons are re-enabled even after an error.

**Data Files**

**ResNorm**
[irs26173_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2491289/irs26173_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2491290/irs26173_graphite002_res.zip)

**Quasi**
[irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2491291/irs26176_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2491292/irs26173_graphite002_res.zip)

**Stretch**
[irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2491291/irs26176_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2491292/irs26173_graphite002_res.zip)

Fixes #23827 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
